libraries/common/array.effekt

```diff
@@ -4,24 +4,52 @@ import effekt
 import exception
 import list
 
-// Primitive operations:
-
 /**
  * A mutable 0-indexed fixed-sized array.
  */
 extern type Array[T]
 
 /**
- * Violated Postcondition: Fields contain values
+ * Allocates a new array of size `size`, keeping its values _undefined_.
+ * Prefer using `array` constructor instead to ensure that values are defined.
  */
-extern global def unsafeFresh[T](size: Int): Array[T] =
+extern global def allocate[T](size: Int): Array[T] =
   js "(new Array(${size}))"
   chez "(make-vector ${size})" // creates an array filled with 0s on CS
   llvm """
     %z = call %Pos @c_array_new(%Int %size)
     ret %Pos %z
   """
 
+/**
+ * Creates a new Array of size `size` filled with the value `init`
+ */
+extern global def array[T](size: Int, init: T): Array[T] =
+  ml "Array.array (${size}, ${init})"
+  default {
+    val arr = allocate[T](size);
+    each(0, size) { i =>
+      unsafeSet(arr, i, init)
+    };
+    arr
+  }
```

Comment:
Is this more efficient on ml this way? Or is this even mandatory?

Reply:
I didn't add this code, I just changed the order of functions in that file, sorry. :)